service/history/task/cross_cluster_task_processor.go

```diff
@@ -45,6 +45,7 @@ const (
 type (
 	// CrossClusterTaskProcessorOptions configures crossClusterTaskProcessor
 	CrossClusterTaskProcessorOptions struct {
+		Enabled                    dynamicconfig.BoolPropertyFn
```

Comment:
I think this property should not be part of dynamic config. If you set this property to true, the service won't immediately start processing cross cluster tasks.
Application of this config will required restart/redeploy.

Also, restart of the service can crash/fail if the config was enabled outside of the service deployment.

Reply:
Open to changing it to be a plain static config value, but this is not significantly different to `history.crossClusterFetcherParallelism` which is currently being used to reduce the overhead of this feature in prod. Both of these values require restart.

Reply:
I guess it is worth discussing on what we want to settle on. I'm not sure that this approach is good, maybe it is worth updating before cross cluster has mass adoption?

Also, I would propose to unify crosscluster configuration at least logically in the code. I see that we have crossClusterProcessor and crossClusterTaskProcessors. One is configured and optional, another is not configured and not optional. Can they be unified somehow and started only if they are required?

Reply:
At this point I'm not super sure we won't just deprecate this feature as it's not really had wide utility and the load cost is pretty high. I would be happy to do such refactoring, but would want to also put a lot more energy into looking into a more lightweight replication mechanism rather than shard polling (pull-based), as this is n^2 number of polls required